src/pip/_internal/locations/__init__.py

```diff
@@ -45,7 +45,20 @@
 
 _PLATLIBDIR: str = getattr(sys, "platlibdir", "lib")
 
-_USE_SYSCONFIG = sys.version_info >= (3, 10)
+
+def _should_use_sysconfig() -> bool:
+    """
+    This function determines the value of _USE_SYSCONFIG.
+    By default, pip uses sysconfig on Python 3.10+.
+    But Python distributors can override this decision by setting:
+        sysconfig._PIP_USE_SYSCONFIG = True / False
+    Rationale in https://github.com/pypa/pip/issues/10647
+    """
+    return bool(getattr(sysconfig, "_PIP_USE_SYSCONFIG", sys.version_info >= (3, 10)))
```

Comment:
I wanted to unwrap this oneliner, but mypy complained (`[attr-defined]`) when I did:

```python
if hasatr(...):
    return bool(sysconfig._PIP_USE_SYSCONFIG)
```

I also considered returning `_PIP_USE_SYSCONFIG` as is without casting it to bool, but this function would have to return `Any` which I didn't like.

Reply:
Yeah Mypy can be dump reading this kind of things. Something like this may work

```python
try:
    use_sysconfig: bool = getattr(sysconfig, "_PIP_USE_SYSCONFIG")
except AttributeError:
    use_sysconfig = sys.version_info >= (3, 10)
return use_sysconfig
```

Reply:
That might work, but feels weird to me, as in the wast majority of cases, the attribute is not there. Hence I would not naturally handle the non-exceptional case as an exception.

Reply:
Just use a `# type: ignore [attr-defined]`. I think that's good enough.

Reply:
These are tools to help us write better code, and disabling them when they get in the way is reasonable.

Reply:
Done.